### ISSUE TITLE
Uncatchable error after repeated calls to Deno.statSync

### ISSUE THREAD

**Issue by 0f-0b** [denoland/deno#16979]
```js
for (let i = 0;; i++) {
  console.log(i);
  Deno.statSync("file that exists");
  try { Deno.statSync("file that does not exist"); } catch {}
}
```

One would expect this loop to continue indefinitely, since any error thrown by `Deno.statSync` should be caught by the surrounding `try` statement. However, it eventually terminates with an uncaught error.

```shellsession
$ touch 'file that exists'
$ deno run --allow-read a.js
0
1
2
…
4079
4080
4081
error: Uncaught NotFound: No such file or directory (os error 2), stat 'file that does not exist'
  try { Deno.statSync("file that does not exist"); } catch {}
             ^
    at Object.statSync (deno:runtime/js/30_fs.js:322:9)
    at file:///cwd/a.js:4:14
$ deno --version
deno 1.28.3+791e623 (canary, x86_64-apple-darwin)
v8 10.9.194.5
typescript 4.8.3
```

**Comment by sigmaSd** [comment]
Can't reproduce on linux

Maybe strace can give you more info

**Comment by 0f-0b** [comment]
> Can't reproduce on linux

Are you using the latest canary version? I can reproduce this in a GitHub Codespace with `deno 1.28.3+a6b5d05 (canary, x86_64-unknown-linux-gnu)`.

**Comment by lino-levan** [comment]
This feels vaguely like https://github.com/denoland/deno/issues/14240 in that the error is uncatchable. I'm wondering if there isn't a root-cause reason for all of these issues.

**Comment by msftceo** [comment]
Getting this error on linux, deno 1.29 with Deno.lstatSync
No error going back to previous release deno 1.28.3.

**Comment by bartlomieju** [comment]
The only commit that might have impact on that is https://github.com/denoland/deno/commit/79285fa83bfbbef55d8afa8f28d11ae4a0b21927. @msftceo any chance you could try to repeat it with that commit and https://github.com/denoland/deno/commit/a7dd28a07c3d018c326636aa563ee4e9200445f8?

**Comment by dsherret** [comment]
Found the issue. Going to do a patch release as soon as it lands.

**Comment by dsherret** [comment]
This is fixed in 1.29.1
